src/server/transaction.cc

```diff
@@ -490,15 +489,22 @@ bool Transaction::RunInShard(EngineShard* shard, bool txq_ooo) {
   }
 
   /*************************************************************************/
+  // at least the coordinator thread owns the reference.
+  DCHECK_GE(GetUseCount(), 1u);
+
+  shard->db_slice().OnCbFinish();
+
```

Comment:
No clue what this is, but it's called on every hop unconditionally (even if we don't conclude)

Reply:
@adiholden  did you mean calling this on every hop even if a tx does not conclude?

Reply:
yes every hop